clients/src/test/java/org/apache/kafka/clients/consumer/CooperativeStickyAssignorTest.java

```diff
@@ -116,6 +130,74 @@ public void testAllConsumersHaveOwnedPartitionInvalidatedWhenClaimedByMultipleCo
         assertTrue(isFullyBalanced(assignment));
     }
 
+    @Test
+    public void testMemberDataWithInconsistentData() {
+        Map<String, Integer> partitionsPerTopic = new HashMap<>();
+        partitionsPerTopic.put(topic, 2);
+        List<TopicPartition> ownedPartitionsInUserdata = partitions(tp1);
+        List<TopicPartition> ownedPartitionsInSubscription = partitions(tp0);
+
+        assignor.onAssignment(new ConsumerPartitionAssignor.Assignment(ownedPartitionsInUserdata), new ConsumerGroupMetadata(groupId, generationId, consumer1, Optional.empty()));
+        ByteBuffer userDataWithHigherGenerationId = assignor.subscriptionUserData(new HashSet<>(topics(topic)));
+        // The owned partitions and generation id are provided in user data and different owned partition is provided in subscription without generation id
+        // If subscription provides no generation id, we'll honor the generation id in userData and owned partitions in subscription
+        Subscription subscription = new Subscription(topics(topic), userDataWithHigherGenerationId, ownedPartitionsInSubscription);
+
+        AbstractStickyAssignor.MemberData memberData = memberData(subscription);
+        // In CooperativeStickyAssignor, we'll serialize owned partition in subscription into userData
```

Comment:
Is this sentence correct? In CooperativeStickyAssignor, only the generation is put in user data, no?

Reply:
You're right! In CooperativeStickyAssignor, only the generation is put in user data. Updated to:
`// In CooperativeStickyAssignor, we only serialize generation id into userData`